cmd/docs.go

```diff
@@ -67,24 +97,145 @@ func docsMain(cmd *cobra.Command, args []string) exitCode {
 			}
 		}
 		if len(supportedSignals) > 0 {
-			printSignalSet("SUPPORTED SIGNALS", supportedSignals)
+			addSection(docs, stringifySignalSet("SUPPORTED SIGNALS", supportedSignals))
 		}
 		if len(supportedSignalGroups) > 0 {
-			printSignalSet("SUPPORTED SIGNAL GROUPS", supportedSignalGroups)
+			addSection(docs, stringifySignalSet("SUPPORTED SIGNAL GROUPS", supportedSignalGroups))
 		}
 	}
 
+	cmdutil.Println(docs.String())
 	return exitCode{0}
 }
 
-func printSignalSet(setName string, signals []apitypes.SignalSchema) {
-	cmdutil.Println()
-	cmdutil.Printf("%v\n", setName)
+func stringifySupportedAttributes(path string, entry apitypes.Entry) string {
+	var supportedAttributes strings.Builder
+	supportedAttributes.WriteString("SUPPORTED ATTRIBUTES\n")
+	for attr, value := range entry.Attributes.ToMap(false) {
+		supportedAttributes.WriteString(fmt.Sprintf("* %v", attr))
+		var fullAttrName string
+		switch attr {
+		case "atime":
+			fullAttrName = "last access time"
+		case "mtime":
+			fullAttrName = "last modified time"
+		case "ctime":
+			fullAttrName = "change time"
+		case "crtime":
+			fullAttrName = "creation time"
+		}
+		if len(fullAttrName) > 0 {
+			supportedAttributes.WriteString(fmt.Sprintf(" (%v)", fullAttrName))
+		}
+		supportedAttributes.WriteString(fmt.Sprintf(" -- %s\n", value))
+	}
+	supportedAttributes.WriteString("\n")
+	metadataLines := []string{
+		fmt.Sprintf("The attributes are a subset of the entry's metadata, which contains everything"),
+		fmt.Sprintf("you ever need to know about the entry. You can use"),
+		fmt.Sprintf("    meta %s", path),
+		fmt.Sprintf("to view the metadata and"),
+		fmt.Sprintf("    meta --attribute %s", path),
+		fmt.Sprintf("to view the (partial) metadata."),
+	}
+	supportedAttributes.WriteString(strings.Join(metadataLines, "\n"))
+	return supportedAttributes.String()
+}
+
+func stringifySupportedActions(path string, entry apitypes.Entry) string {
+	var supportedActions strings.Builder
+	supportedActions.WriteString("SUPPORTED ACTIONS\n")
+	actions := entry.Actions
+	sort.Strings(actions)
+	for _, action := range actions {
+		supportedActions.WriteString(fmt.Sprintf("* %v\n", action))
+		var actionDescriptionLines []string
+		switch action {
+		case plugin.ListAction().Name:
+			actionDescriptionLines = []string{
+				fmt.Sprintf("- ls %s", path),
+				fmt.Sprintf("    Type 'docs <child>' to view an ls'ed child's documentation"),
+				fmt.Sprintf("- cd %s", path),
+				fmt.Sprintf("    The 'W' environment variable contains the Wash root. Use 'cd $W' to"),
+				fmt.Sprintf("    return to it."),
+				fmt.Sprintf("- stree %s", path),
+				fmt.Sprintf("    Gives you a high-level overview of the kinds of things that you will"),
+				fmt.Sprintf("    encounter when you 'cd' and 'ls' through this entry."),
+				fmt.Sprintf("- find %s <expression>", path),
```

Comment:
Thinking of moving over `find`'s introduction to the "SUPPORTED ATTRIBUTES" section since attributes/metadata filtering is probably `find`'s biggest use-case.

Reply:
Seems reasonable.